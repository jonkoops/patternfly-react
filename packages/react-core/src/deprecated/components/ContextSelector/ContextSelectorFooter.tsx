import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/ContextSelector/context-selector';

export interface ContextSelectorFooterProps extends HTMLProps<HTMLDivElement> {
  /** Content rendered inside the ContextSelectorFooter */
  children?: ReactNode;
  /** Additional classes added to the ContextSelectorFooter */
  className?: string;
}

export const ContextSelectorFooter: FunctionComponent<ContextSelectorFooterProps> = ({
  children = null,
  className = '',
  ...props
}: ContextSelectorFooterProps) => (
  <div {...props} className={css(styles.contextSelectorMenuFooter, className)}>
    {children}
  </div>
);
ContextSelectorFooter.displayName = 'ContextSelectorFooter';
