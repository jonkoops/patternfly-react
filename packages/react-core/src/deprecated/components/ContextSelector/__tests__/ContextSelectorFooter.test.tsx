/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { expect, test } from 'vitest';

import { ContextSelectorFooter } from '../ContextSelectorFooter';

test('Renders ContextSelectorFooter', () => {
  const { asFragment } = render(<ContextSelectorFooter>testing text</ContextSelectorFooter>);
  expect(asFragment()).toMatchSnapshot();
});
