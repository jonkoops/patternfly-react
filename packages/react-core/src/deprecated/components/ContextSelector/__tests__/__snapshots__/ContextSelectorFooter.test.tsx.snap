// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders ContextSelectorFooter 1`] = `
<DocumentFragment>
  <div
    class="pf-v5-c-context-selector__menu-footer"
  >
    testing text
  </div>
</DocumentFragment>
`;
